elastic/kibana
# migrate es query rule tests to the deployment agnostic framework

Fixes https://github.com/elastic/kibana/issues/183395

This PR migrates the ES query rule tests to the [deployment agnostic framework](https://github.com/elastic/kibana/tree/main/x-pack/test/api_integration/deployment_agnostic)

### TODO

- [x] Migrate ES rule tests into the deployment agnostic solution
- [x] Test in MKI before merging

### How to run tests locally

To run serverless
```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts --grep="ElasticSearch query rule"
```

To run stateful
```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts --grep="ElasticSearch query rule"
```

### How to run tests on MKI

According to this [discussion](https://github.com/elastic/observability-dev/issues/3519#issuecomment-2379914274), we should test in MKI environment before merging. For details on how to run in MKI, see [this section of the document](https://docs.google.com/document/d/1tiax7xoDYwFXYZjRTgVKkVMjN-SQzBWk4yn1JY6Z5UY/edit#heading=h.ece2z8p74izh) and [this readme](https://github.com/elastic/kibana/blob/main/x-pack/test_serverless/README.md#run-tests-on-mki).
